### ISSUE TITLE
Feature: Timeout conditions

### ISSUE THREAD

**Issue by weavejester** [imjp94/gd-YAFSM#40]
I often find myself needing state transitions based on a timeout. For example transitioning a "dash" to "idle" state after 0.1 seconds.

Would you be interested in a patch for a new "timeout" condition type that would trigger after the state has been active for a specified amount of time? I'd personally find it convenient, but I can also understand if you want to keep the FSM as something solely triggered by explicit parameter changes.

**Comment by weavejester** [comment]
I put together a prototype for my own use. The timeout state transition looks like:

![image](https://user-images.githubusercontent.com/8780/152459748-381c690d-8571-499b-a889-79c236bbf496.png)

With the timeout condition editor looking like:

![image](https://user-images.githubusercontent.com/8780/152459807-54913ffb-f8f6-4982-9ce2-2fcc0b93860b.png)

**Comment by imjp94** [comment]
Yes, I think TimeoutCondition can be very useful in many cases, even the gd-YAFSM demo project would benefit from it too.
And your prototype looks really good, would you like to open a pull request?
